core/tests/test_ingress.py

```diff
@@ -0,0 +1,237 @@
+
+    # Creates/Opens a new connection to the db and binds the engine
+    session = session_open(engine)
+
+    # Check if table is empty and bulk inserts if it is
+    first_entry = session.query(SensorLocationClass).first()
+    if first_entry == None:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        assert session.query(SensorLocationClass).count() == len(sensor_df.index)
+    else:
+        assert session.query(SensorLocationClass).count() == len(sensor_df.index)
+
+    session_close(session)
+
+     # Trying to upload location history data for a sensor that does not exist
+    test_csv = "sensor_location_test_1.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
+
+    # Trying to upload location history data for a location that does not exist
+    test_csv = "sensor_location_test_2.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
+
+    # Trying to upload location history data with an empty installation date
+    test_csv = "sensor_location_test_3.csv"
+
+    sensor_df = pd.read_csv(os.path.join(CONST_TEST_DIR_DATA, CONST_SENSOR_LOCATION_TESTS, test_csv))
+    assert sensor_df.empty == False
+    status, log, engine = connect_db(SQL_CONNECTION_STRING, TEST_DB_NAME)
+    assert status, log
+    session = session_open(engine)
+
+    try:
+        session.bulk_insert_mappings(SensorLocationClass, sensor_df.to_dict(orient="records"))
+        result = True
+    except:
+        session.rollback()
+        result = False
+
+    assert result == False
+
+    session_close(session)
+
+def test_import_data ():
+
+    # Bring df
+    success, log, test_ingress_df = advanticsys_import(file_path)
+    assert success, log
+    assert isinstance(test_ingress_df, pd.DataFrame)
+
+    # Test import function
+    success, log = import_data (
+        test_ingress_df,
+        SENSOR_TYPE,
+        SQL_SERVER,
+        SQL_USER,
+        SQL_PASSWORD,
+        SQL_HOST,
+        SQL_PORT,
+        TEST_DB_NAME
+    )
+    assert success, log
+
```

Comment:
Are we sure that this test is enough to test all the boundary, missing or invalid data cases, or are these test are performed elsewhere?

Reply:
most are performed elsewhere (missing/invalid --> test_ingress.py), not sure what boundary is,

Reply:
This function will be the key function in the import process. We need to make sure that it works in every scenario, i.e. no data, no df, incorrect SQL details, wrong data in the df, incomplete data in df, duplicate data in df, wrong format data in df, wrong df format. We need to tests all the scenarios. At the moment this test is only for one scenario when everything goes smoothly.